### ISSUE TITLE
Batching API

### ISSUE THREAD

**Issue by radike** [vaclavHala/SpatialTracker#16]
The server uses Batching API to:
a) computes some statistics from the data; or b) send emails to users informing them that their issues had been closed

**Comment by OndrejPontes** [comment]
@radike @vaclavHala Batching je konecne funkcny a taktiez heat mapy, zajtra zrefaktorujem kod a mergnem s mastrom, zatial to je len v Infinispan branchy, ak ste uz niekedy robili batching tak by ste sa mohli pozriet ci sa to da spravit lepsie

**Comment by ivan-kralik** [comment]
Čo je potrebné urobiť pre fungovanie batching API? Prečo má trieda `cz.muni.fi.pv243.spatialtracker.infinispan.Resource` anotáciu `@RolesAllowed("USER")` a jediná metóda, čo tam je, tak má `@PermitAll`? Prečo sa má pri volaní `createHeatMap()` používať HTTP metóda `POST`, keď táto metóda vracia heat mapu a nič na serveri nemení (ak som to správne pochopil)? Nebolo by vhodnejšie použiť `GET`?

Keď zavolám:

`curl -v -XPOST 'http://localhost:8080/tracker/rest/cache/heat_map?filter=\[\]'`

Na serveri nastane sa vyhodí výnimka:

`EJB Invocation failed on component Resource for method public java.util.List cz.muni.fi.pv243.spatialtracker.infinispan.Resource.createHeatMap(java.lang.String): javax.ejb.EJBAccessException: WFLYEJB0364: Invocation on method: public java.util.List cz.muni.fi.pv243.spatialtracker.issues.IssueResource.getFilters(java.lang.String) throws java.io.IOException of bean: IssueResource is not allowed`

**Comment by OndrejPontes** [comment]
Batching nabehne hneď po deploznuti, to s tymi anotaciami som si nevsimol, opravim to, a na tie heat mapy sa pozriem hned ako budem pri ntbku

**Comment by ivan-kralik** [comment]
Stále, keď zavolám príkaz:

`curl -v 'http://localhost:8080/tracker/rest/heat_map?filter=\[\]' --user admin:admin`

Dostanem chybový kód 500 a na serveri sa vyhodí výnimka:

`com.fasterxml.jackson.databind.JsonMappingException: Unexpected token (END_ARRAY), expected VALUE_STRING: need JSON String that contains type id (for subtype of cz.muni.fi.pv243.spatialtracker.issues.filter.SpatialFilter)
20:27:48,141 ERROR [stderr] (default task-33)  at [Source: []; line: 1, column: 2]`

Vám to funguje?

**Comment by OndrejPontes** [comment]
mne to fungovalo, mas najnovsiu verziu ?

2016-06-06 20:30 GMT+02:00 ivan-kralik notifications@github.com:

**Comment by ivan-kralik** [comment]
Mám branch master, commit `ab91243d75f82c60608f37b01e96417aa9656f3f` a toto mi píše, že je najnovšia verzia. Skúšal si presne ten CURL príkaz?

**Comment by OndrejPontes** [comment]
ja som to skusal cez postmana

2016-06-06 20:35 GMT+02:00 ivan-kralik notifications@github.com:

**Comment by OndrejPontes** [comment]
ale daj mi chvilku a hned ti to potvrdim

Dňa 6. júna 2016, 20:36, Ondrej Ponteš ondrej.pontes@gmail.com
napísal(-a):

**Comment by OndrejPontes** [comment]
ano, vyhodi to aj mne, ale nerozumiem preco, ak tam nekontrolujem aka rola
ma pristup k tej metode tak to funguje spravne,
ale akonahle tam dam podmienku aby to bol user tak to vyhodi vynimku,
skusim to fixnut

Dňa 6. júna 2016, 20:36, Ondrej Ponteš ondrej.pontes@gmail.com
napísal(-a):
